lib/board.rb

```diff
@@ -0,0 +1,47 @@
+require 'pry'
+
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end
+
+  def full?
+    cells.none? { |position| position == " "}
+  end
+
+  def turn_count
+    cells.select { |position| position != " "}.count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```

Reply:
I had no idea .count took arguments! I guess I learned it a while back when only needing to return a number of elements in an array and never went back to look at it in the Docs! so useful!